src/commands/edge_app.rs

```diff
@@ -0,0 +1,241 @@
+        let config = Config::new("asdf".to_owned());
+        let authentication = Authentication::new_with_config(config);
+        let command = EdgeAppCommand::new(authentication);
+
+        let p = tmp_dir.path().join("screenly.yml");
+        assert!(command.init(Path::new(p.to_str().unwrap())).is_ok());
+
+        assert_eq!(
+            EdgeAppManifest::default(),
+            serde_yaml::from_str(&fs::read_to_string(Path::new(p.to_str().unwrap())).unwrap())
+                .unwrap()
+        );
+    }
+
+    #[test]
+    fn test_edge_app_publish_should_send_correct_request() {
+        let tmp_dir = TempDir::new("test").unwrap();
+        let _lock = lock_test();
+        let _test = set_env(OsString::from("HOME"), tmp_dir.path().to_str().unwrap());
+        fs::write(tmp_dir.path().join(".screenly").to_str().unwrap(), "token").unwrap();
+        let manifest = EdgeAppManifest {
+            id: "".to_string(),
+            created_at: "".to_string(),
+            created_by: "".to_string(),
+            permissions: vec![],
+            updated_at: "".to_string(),
+
+            name: "Test".to_string(),
+            version: "100".to_string(),
+            description: "Best".to_string(),
+            icon: "?".to_string(),
+            author: "Best author ever".to_string(),
+            homepage_url: "test.io".to_string(),
+        };
+
+        let published_manifest = vec![EdgeAppManifest {
+            id: "01GS5H2CX6Y10ZRJHEDQPEWN4E".to_string(),
+            created_at: "2023-02-13T13:57:43+00:00".to_string(),
+            created_by: "01D0C6YJA8000A7JEN0230MWGF".to_string(),
+            updated_at: "2023-02-13T13:57:43.97356+00:00".to_string(),
+            permissions: vec![],
+
+            ..manifest.clone()
+        }];
+
+        let mut b = serde_json::to_value(&manifest).unwrap();
+        let manifest_object = b.as_object_mut().unwrap();
+        manifest_object.remove("id");
+        manifest_object.remove("created_at");
+        manifest_object.remove("created_by");
+        manifest_object.remove("updated_at");
+        manifest_object.remove("permissions");
+
+        let mock_server = MockServer::start();
+        mock_server.mock(|when, then| {
+            when.method(POST)
+                .path("/v4/edge_apps")
+                .header("Authorization", "Token token")
+                .header(
+                    "user-agent",
+                    format!("screenly-cli {}", env!("CARGO_PKG_VERSION")),
+                )
+                .json_body(serde_json::to_value(manifest_object).unwrap());
+
+            then.status(201)
+                .json_body(serde_json::to_value(&published_manifest).unwrap());
+        });
+
+        let config = Config::new(mock_server.base_url());
+        let authentication = Authentication::new_with_config(config);
+        let command = EdgeAppCommand::new(authentication);
+
+        let p = tmp_dir.path().join("screenly.yml");
```

Comment:
Use meaningful variable names please.

Reply:
again - this is test code - why does "p" not work?